adagrams/game.py

```diff
@@ -1,11 +1,131 @@
+import random
+import array
+
+# manually typed list of all the letters (and how many of them as repeats) that are allowed
+LETTER_LIST = ['A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'B', 'B', 'C', 'C', 'D', 'D', 'D', 'D', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'F', 'F', 'G', 'G', 'G', 'H', 'H', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'J', 'K', 'L', 'L', 'L', 'L', 'M', 'M', 'N', 'N', 'N', 'N', 'N', 'N', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'P', 'P', 'Q', 'R', 'R', 'R', 'R', 'R', 'R', 'S', 'S', 'S', 'S', 'T', 'T', 'T', 'T', 'T', 'T', 'U', 'U', 'U', 'U', 'V', 'V', 'W', 'W', 'X', 'Y', 'Y', 'Z']
+
 def draw_letters():
-    pass
+    letters = [] # empty 'letters' list 
+
+    letters = random.sample(LETTER_LIST,10) #10 letters are randomly selected and set to the letters list
+    
+    return letters #returns 10 randomly selected letters 
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    #loops through all the letters in the words and makes them lowercase strings in a new list
+    word_case_insensitive =  []
+    for i in word:
+        word_case_insensitive.append(i.lower())
+
+    #loops through all the letters in the letter bank and makes them lowercase strings in a new list
+    letter_bank_case_insensitive = []
+    for i in letter_bank:
+        letter_bank_case_insensitive.append(i.lower())
+
+    #counter for how many loops are completed
+    loop_count = 0
+
+    #for-loop that compares each item in both lists, and removes from letter bank if there's a match
+    for i in word_case_insensitive:
+        for j in letter_bank_case_insensitive:
+            if i == j:
+                loop_count += 1
+                letter_bank_case_insensitive.remove(j)
+
+    #if the loop count and length of a word are equal, it returns True and else False
+    if loop_count == len(word):
+        return True
+    else:
+        return False
+        
+            
+
 
 def score_word(word):
-    pass
+    #lists of scores grouped by number of points. Upper and lower are included to cover all bases (even though it might be not needed since the letters in the word are turned into lowercase)
```

Comment:
This covers all the possibilities, but it would be ideal to programmatically check for letter case instead of hard-coding both strings.